CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(FLAC VERSION 1.3.2) # HOMEPAGE_URL "https://www.xiph.org/flac/")
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(BUILD_CXXLIBS "Build libFLAC++" ON)
+option(BUILD_EXAMPLES "Build and install examples" ON)
+option(WITH_OGG "ogg support (default: test for libogg)" ON)
+
+if(WITH_OGG)
+    find_package(OGG REQUIRED)
+endif()
+
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes -Waggregate-return -Wcast-align -Wnested-externs -Wshadow -Wundef -Wmissing-declarations -Winline")
+
+    if(CMAKE_BULD_TYPE STREQUAL Release)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -funroll-loops")
```

Comment:
Setting optimization flags is tricky. If you run cmake without `CMAKE_BUILD_TYPE`, `CMAKE_C_FLAGS` is empty you can add "-O3" safely. But if `CMAKE_BUILD_TYPE` is set, `CMAKE_C_FLAGS` is not empty (`-O2 -DNDEBUG`) and you will not get what you want.

Reply:
From GCC manual:
> If you use multiple -O options, with or without level numbers, the last such option is the one that is effective.

Reply:
Checked right now, with release build type:

```
//Flags used by the C compiler during DEBUG builds.
CMAKE_C_FLAGS_DEBUG:STRING=-g

//Flags used by the C compiler during MINSIZEREL builds.
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG

//Flags used by the C compiler during RELEASE builds.
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG

//Flags used by the C compiler during RELWITHDEBINFO builds.
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
```

CMake just ignores your settings.

Reply:
I've changed CMAKE_C_FLAGS to CMAKE_C_FLAGS_RELEASE. Now gcc gets correct flags

Reply:
`-O3` twice and only useful `-funroll-loops` option added?

Add `-funroll-loops` with `target_compile_options` as private. Don't touch `-Ox` options. `-O3` is not suitable sometimes (my linux distros wants -O2 as safe default).

`-Ox` switch can be overriden from command line or with toolchain file (` CMAKE_<LANG>_FLAGS_<CONFIG>_INIT`). But if you modify something inside your `CMakeLists.txt` it cannot be undone.

Reply:
If there is a practical need for this it can be changed afterwards. As to this PR I wrote it like this because configure.ac has this commands. If it really worries you you can propose a patch to this later and discuss it with the project owner.